pkg/cloud/scope/machine.go

```diff
@@ -180,11 +180,9 @@ func (m *MachineScope) Close() error {
 // AdditionalTags merges AdditionalTags from the scope's AWSCluster and AWSMachine. If the same key is present in both,
 // the value from AWSMachine takes precedence. The returned Tags will never be nil.
 func (m *MachineScope) AdditionalTags() infrav1.Tags {
-	var tags infrav1.Tags = m.AWSCluster.Spec.AdditionalTags
-	if tags == nil {
-		tags = infrav1.Tags{}
-	}
+	var tags infrav1.Tags = make(infrav1.Tags)
 
+	tags.Merge(m.AWSCluster.Spec.AdditionalTags)
 	tags.Merge(m.AWSMachine.Spec.AdditionalTags)
```

Comment:
The above lines have such a subtle difference is almost invisible 👻, can we add a comment above both?